peps/pep-0746.rst

```diff
@@ -43,15 +43,14 @@ This use case comes up in libraries like :pypi:`pydantic`, which use
 
 Specification
 =============
-
-This PEP introduces a new ``__supports_type__`` protocol that both static and
-runtime type checkers can use to understand if a metadata object in
-``Annotated`` is valid for the given type. Objects that implement this protocol
-must have a method named ``__supports_type__`` that takes a single positional argument and
-returns ``bool``::
+This PEP introduces a protocol that can be used by static and runtime type checkers to validate
+the consistency between Annotated metadata and a given type.
+Objects that implement this protocol have a method named ``__supports_type__``
+that takes a single positional-only argument (denoted by a ``, /`` in the function signature)
+and returns ``bool``::
```

Comment:
I agree with making the argument positional-only in the `__supports_type__` protocol definition given in the PEP. But I'm not sure we should specify that the argument _has_ to be positional-only in _implementers_ of the protocol. In general, due to contravariant subtyping of parameters, a signature with a positional-or-keyword parameter is always considered a subtype of a signature with a positional-only parameter, since the signature with the positional-or-keyword parameter is strictly more permissive:

```py
from typing import Protocol

class Foo(Protocol):
    def __call__(self, x: int, /) -> str: ...

def bar(y: int) -> str:
    return "bar" * y

baz: Foo = bar  # this is fine
```

Reply:
Does just changing the wording to `positional-only` but dropping the `, /` part sound good then?

Reply:
Yeah, or even just "that takes a single _positional_ argument" (rather than _positional-only_)

Reply:
I interpreted Eric's comment as saying that the protocol _definition_ given in the PEP should use a positional-only parameter (since we don't want to mandate a specific parameter name for implementers of the protocol), not that we should mandate that implementers of the protocol _have_ to use a positional-only parameter. But yeah, let's wait for Eric to chime in :-)

Reply:
I also don't think this should say "positional-only": the specification should be that it accepts a positional argument, but it is not necessary to accept *only* a positional argument.

Reply:
I think we're confusing arguments and parameters here. The _argument_ (the value passed to the `__supports_type__ ` method) must be positional. There's no such thing as a "positional-only argument", so that phrase shouldn't be used.

The protocol definition should indicate that the _parameter_ is positional-only. The implementation of the protocol could use a standard (positional+keyword) parameter as per the [subtyping rules for callables](https://typing.readthedocs.io/en/latest/spec/callables.html#parameter-kinds).

Reply:
There is no explicit protocol definition though, only an example. This PR already adds a `/` to the example; perhaps that is enough?

Reply:
Ah, I see what you mean. It's a bit weird that the PEP doesn't currently spell out the protocol in code form. Wouldn't that make it clearer rather than attempting to describe the protocol in text?

In any case, the current text in this PR is incorrect. The phrase "that takes a single positional-only argument (denoted by a ``, /`` in the function signature)" should probably just say "that takes a single positional argument".

Reply:
> Wouldn't that make it clearer rather than attempting to describe the protocol in text?

So you'd want something like `class TypedMetadata[T](Protocol)` in the PEP (with the caveat that we need to clarify that the code is for description purposes only and won't exist anywhere in real life)?

Reply:
Thank you for the the feedback and education all. I've removed the incorrect wording and instead added a code definition of the protocol. I tried to word it as to make it clear that there is no actual code definition anywhere: `The protocol being introduced would be defined as follows if it were to be defined in code form:`. Open to suggestions to make that clearer.